components/Button/Button.jsx

```diff
@@ -1,91 +1,329 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import styled from 'styled-components';
+import { shadow, hexToRgba } from '../shared';
+import {
+  components,
+  spacing,
+  colors as defaultColors,
+  baseFontSize as defaultBaseFontSize,
+} from '../shared/theme';
 
 import Icon from '../Icon/Icon';
-import Colors from '../Colors';
-import buttonStyle from './buttonStyle';
-
-const iconSize = ({ size }) => {
-  const sizes = {
-    xsmall: '16px',
-    small: '16px',
-    normal: '24px',
-    large: '24px',
-    xlarge: '24px',
-  };
-
-  return `font-size: ${sizes[size] || sizes.normal};`;
-};
 
 const ButtonIcon = styled(Icon)`
   pointer-events: none;
 `;
 
 const StyledButton = styled.button`
-  ${buttonStyle}
+  align-items: center;
+  display: flex;
+  font-weight: bold;
+  justify-content: center;
+  letter-spacing: 0.2px;
+  border-radius: 4px;
+
+  ${props => `cursor: ${props.disabled ? 'not-allowed' : 'pointer'};`}
+
+  ${({ size, theme: { baseFontSize } }) => {
+    const sizes = {
+      xsmall: `${baseFontSize * 0.75}px`,
+      small: `${baseFontSize * 0.75}px`,
+      medium: `${baseFontSize}px`,
+      large: `${baseFontSize * 1.25}px`,
+      xlarge: `${baseFontSize * 1.5}px`,
+    };
+    return `font-size: ${sizes[size] || sizes.medium};`;
+  }}
+
+  ${({
+    size,
+    theme: {
+      spacing: { large, xlarge, xxlarge, xxxlarge },
+    },
+  }) => {
+    const heights = {
```

Comment:
Is this property really needed? Can we solve with padding only?
If can't, IMHO I'd avoid using spacing values for component height

Reply:
_Is this property really needed?_
Yes. It is needed.

_Can we solve with padding only?_
Yes, we can.

_If can't, IMHO I'd avoid using spacing values for component height_
We follow material design (MD) guidelines as a reference for this. In this case, the MD uses height to define the vertical size of the component.

Reply:
If we can solve with padding, so it is not needed (respectin')
I'd remove this part, but i'll leave to you guys to decide or vote <3

Reply:
Although Material Design specifies that this is solved with height, it's a common sense among us (including me) that padding it's better for this case. So, is there any problem to implements in this way?